### ISSUE TITLE
Destroying Groups

### ISSUE THREAD

**Issue by Mahrkeenerh** [kdybicz/adobe-color-swatch#4]
Unfortunately, your solution is ignoring the color groups, therefore it's a destructive workflow.

**Comment by kdybicz** [comment]
@Mahrkeenerh that's true, I didn't figured out hey how Adobe encodes group informations

**Comment by Mahrkeenerh** [comment]
I do not know how it's encoded, but I can read the information:

At the end of the file (after v2), there's some junk, and then `Grup TEXT`, then the name of the group (plus extra padding everywhere). Then there's `Objc preset` which signalizes one color. When the group ends, there's `groupEnd`.

Meaning you can have nested groups, that would be: `TEXT`, `TEXT`, `preset`s, `end` `end`.

**Comment by kdybicz** [comment]
I looked across the Internet and I haven't found anything that would help me to understand the structure of the `8BIM` `phry` extension used for color groups in this file format. There is no official nor unofficial documentation I could use, so for now I need to park any related work.
